modules/src/ics02_client/msgs.rs

```diff
@@ -15,17 +24,155 @@ pub enum ClientMsg {
 }
 
 /// A type of message that triggers the creation of a new on-chain (IBC) client.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq)]
 pub struct MsgCreateAnyClient {
     pub client_id: ClientId,
     pub client_type: ClientType,
     pub client_state: AnyClientState,
     pub consensus_state: AnyConsensusState,
+    pub signer: AccountId,
+}
+
+impl MsgCreateAnyClient {
+    pub fn new(
+        client_id: ClientId,
+        client_type: ClientType,
+        client_state: AnyClientState,
+        consensus_state: AnyConsensusState,
+        signer: AccountId,
+    ) -> Self {
+        MsgCreateAnyClient {
+            client_id,
+            client_type,
+            client_state,
+            consensus_state,
+            signer,
+        }
+    }
+}
+
+impl Msg for MsgCreateAnyClient {
+    type ValidationError = crate::ics24_host::error::ValidationError;
+
+    fn route(&self) -> String {
+        crate::keys::ROUTER_KEY.to_string()
+    }
+
+    fn get_type(&self) -> String {
+        TYPE_MSG_CREATE_CLIENT.to_string()
+    }
+
+    fn validate_basic(&self) -> Result<(), Self::ValidationError> {
+        // Nothing to validate since all fields are validated on creation.
+        Ok(())
+    }
+
+    fn get_sign_bytes(&self) -> Vec<u8> {
+        let mut buf = Vec::new();
+        let raw_msg: MsgCreateClient = self.clone().into();
+        prost::Message::encode(&raw_msg, &mut buf).unwrap();
+        buf
+    }
+
+    fn get_signers(&self) -> Vec<AccountId> {
+        vec![self.signer]
+    }
+}
+
+impl DomainType<MsgCreateClient> for MsgCreateAnyClient {}
+
+impl TryFrom<MsgCreateClient> for MsgCreateAnyClient {
+    type Error = Error;
+
+    fn try_from(raw: MsgCreateClient) -> Result<Self, Self::Error> {
+        let raw_client_state = match raw.client_state {
+            None => Err(Kind::DecodeMessage.context(error::Kind::InvalidRawClientState)),
+            Some(raw_state) => Ok(raw_state),
+        }?;
+
+        let client_type = match raw_client_state.type_url.as_str() {
+            "/ibc.tendermint.ClientState" => Ok(ClientType::Tendermint),
+
+            _ => Err(
+                Kind::DecodeMessage.context(error::Kind::UnknownConsensusStateType(
+                    raw_client_state.clone().type_url,
+                )),
+            ),
+        }?;
+        let raw_consensus_state = match raw.consensus_state {
+            None => Err(Kind::DecodeMessage.context(error::Kind::InvalidRawConsensusState)),
+            Some(raw_consensus_state) => Ok(raw_consensus_state),
+        }?;
+
+        Ok(MsgCreateAnyClient {
+            client_id: raw.client_id.parse().unwrap(),
+            client_type,
+            client_state: AnyClientState::try_from(raw_client_state).unwrap(),
+            consensus_state: AnyConsensusState::try_from(raw_consensus_state).unwrap(),
+            signer: AccountId::new(<[u8; 20]>::try_from(&raw.signer[..20]).unwrap()),
```

Comment:
Something like this should work as well:

```suggestion
            signer: AccountId::new(raw.signer[..20].try_into()).unwrap()),
```

Perhaps this could be added as a static method on `AccountId`?

```rust
fn from_bytes(bytes: &[u8]) -> Result<Self, std::array::TryFromSliceError>> {
  Ok(Self::new(raw.signer[..20].try_into()?))
}
```

I now see that the type lives in `tendermint-rs` so we can definitely leave this up as a future improvement.